src/App.jsx

```diff
@@ -14,45 +17,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_NAME = 'alph';
+const SORT_BY_LENGTH = 'length';
+
+function sortGoods(goods, { sortType, isReversed }) {
+  const copyOfGoods = [...goods];
+
+  copyOfGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SORT_BY_NAME:
+        return good1.localeCompare(good2);
+      case SORT_BY_LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    copyOfGoods.reverse();
+  }
+
+  return copyOfGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState('');
+  const sortedGoods = sortGoods(goodsFromServer, { sortType, isReversed });
+  const resetFilter = () => {
+    setReversed(false);
+    setSortType('');
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            'button is-info',
+            { 'is-light': sortType !== SORT_BY_NAME },
+          )}
+          onClick={() => {
+            setSortType(SORT_BY_NAME);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button is-success',
+            { 'is-light': sortType !== SORT_BY_LENGTH },
+          )}
+          onClick={() => {
+            setSortType(SORT_BY_LENGTH);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button is-warning',
+            { 'is-light': isReversed === false },
+          )}
+          onClick={() => setReversed(!isReversed)}
```

Comment:
```suggestion
          onClick={() => setReversed(prevState => !prevState)}
```
Better toggle the state like this